File: cli/src/commands/session/restore.ts
import { Command } from "commander";
import { daemonPost } from "../../lib/daemon-client.js";
import { preflight } from "../../lib/preflight.js";
import { die, success } from "../../lib/output.js";
import ora from "ora";

export function registerSessionRestore(session: Command): void {
  session
    .command("restore <id>")
    .description("Restore a stopped session")
    .action(async (id: string) => {
      await preflight();

      const spinner = ora("Restoring session...").start();

      try {
        const result = await daemonPost<{ id: string }>(
          `/sessions/${id}/resume`
        );

        spinner.stop();

        if (!result.ok) {
          die(result.error, result.status === 404 ? 2 : 1);
        }

        success(`Session restored: ${id}`);
      } catch (err) {
        spinner.fail();
        throw err;
      }
    });
}

File: rust/vst-cli/src/commands/agent/restore.rs
use crate::client::{daemon_post, DaemonResult};
use crate::output::success;
use crate::preflight::preflight;

pub async fn run_session_restore(id: &str) -> Result<(), (String, i32)> {
    if id.is_empty() {
        return Err(("Session ID is required".to_string(), 1));
    }

    preflight().await;

    let result = daemon_post::<serde_json::Value, ()>(&format!("/sessions/{id}/resume"), None)
        .await
        .map_err(|e| (e.to_string(), 1))?;

    match result {
        DaemonResult::Ok { .. } => {
            success(&format!("Session restored: {id}"));
            Ok(())
        }
        DaemonResult::Err { status, error, .. } => {
            let code = if status == 404 { 2 } else { 1 };
            Err((error, code))
        }
    }
}
